File: e-commerce-frontend/src/types/domains/category.ts
import { Attribute } from "./attribute";
import { Variation } from "./variation";

export interface Category {
   categoryId: number;
   name: string;
   code: string;
   parentCategory?: Category;
   variations: Variation[];
   attributes: Attribute[];
   imageUrl?: string;
}

export interface CategoryTree {
   categoryId: number;
   name: string;
   path: string;
   subcategories: CategoryTree[];
   imageUrl?: string;
}

export interface CategoryDetails {
   categoryId: number;
   name: string;
   code: string;
   parentCategory?: CategoryDetails;
   variations: VariationDTO[];
   attributes: Attribute[];
   imageUrl?: string;
}

export interface VariationDTO {
   variationId: number;
   name: string;
}

export interface CategoryDTO {
   name: string;
   code: string;
   parentCategoryId?: number;
   variationIds: number[];
   attributeIds: number[];
   imageUrl?: string;
}

File: e-commerce-backend/src/main/java/com/TechKun/model/Category.java
package com.TechKun.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Data
@NoArgsConstructor
@Table(name = "Category")
public class Category {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "category_id", nullable = false)
    private Integer categoryId;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "name_tr")
    private String nameTr;

    @Column(name = "code", nullable = false, unique = true)
    private String code;

    @Column(name = "image_url")
    private String imageUrl;

    @ManyToOne
    @JoinColumn(name = "parent_id")
    private Category parentCategory;

    @ManyToMany()
    @JoinTable(name = "category_variation",
        joinColumns = @JoinColumn(name = "category_id"),
        inverseJoinColumns = @JoinColumn(name = "variation_id")
    )
    private List<Variation> variations;

    @ManyToMany()
    @JoinTable(name = "category_attribute",
        joinColumns = @JoinColumn(name = "category_id"),
        inverseJoinColumns = @JoinColumn(name = "attribute_id")
    )
    private List<Attribute> attributes;

    public Category(Integer categoryId) {
        this.categoryId = categoryId;
    }
}
